tsl/test/expected/split_chunk.out

```diff
@@ -0,0 +1,383 @@
+------------------+------------------------------+------------------------------
+ _hyper_1_1_chunk | Wed Jan 03 16:00:00 2024 PST | Thu Jan 04 00:00:00 2024 PST
+ _hyper_1_3_chunk | Thu Jan 04 00:00:00 2024 PST | Sun Jan 07 08:00:00 2024 PST
+ _hyper_1_4_chunk | Sun Jan 07 08:00:00 2024 PST | Wed Jan 10 16:00:00 2024 PST
+(3 rows)
+
+-- Check that the counts in the two result partitions is the same as
+-- in the original partition and that the tuple counts are roughly the
+-- same across the partitions.
+with counts as (
+    select (select count(*) from _timescaledb_internal._hyper_1_3_chunk) count1,
+            (select count(*) from _timescaledb_internal._hyper_1_4_chunk) count2
+) select
+  c.count1, c.count2,
+  c.count1 + c.count2 as total_count,
+  (c.count1 + c.count2) = :orig_count as is_same_count
+from counts c;
+ count1 | count2 | total_count | is_same_count 
+--------+--------+-------------+---------------
+  28800 |  23402 |       52202 | t
+(1 row)
+
+-- Check that both rels return proper data and no columns are messed
+-- up
+select time, device, location, temp from _timescaledb_internal._hyper_1_3_chunk order by time, device limit 3;
+             time             | device | location |       temp       
+------------------------------+--------+----------+------------------
+ Thu Jan 04 00:00:00 2024 PST |      2 |        1 | 25.6730424335366
+ Thu Jan 04 00:00:10 2024 PST |      6 |        1 | 15.0341761730165
+ Thu Jan 04 00:00:20 2024 PST |      6 |        1 | 27.5663547962209
+(3 rows)
+
+select time, device, location, temp from _timescaledb_internal._hyper_1_4_chunk order by time, device limit 3;
+             time             | device | location |       temp       
+------------------------------+--------+----------+------------------
+ Sun Jan 07 08:00:00 2024 PST |     10 |        1 | 4.03503358112434
+ Sun Jan 07 08:00:10 2024 PST |      8 |        1 |  17.726969596003
+ Sun Jan 07 08:00:20 2024 PST |      6 |        1 | 9.63191118430237
+(3 rows)
+
+--
+-- Test split with integer time
+--
+create table splitme_int (time int not null, device int, temp float);
+select create_hypertable('splitme_int', 'time', chunk_time_interval => 10::int);
+    create_hypertable     
+--------------------------
+ (3,public,splitme_int,t)
+(1 row)
+
+insert into splitme_int values (1, 1, 1.0), (8, 8, 8.0);
+select ch as int_chunk from show_chunks('splitme_int') ch limit 1 \gset
+select * from chunk_slices;
+ hypertable_name |    chunk_name    |   range_start    |    range_end     
+-----------------+------------------+------------------+------------------
+ splitme_int     | _hyper_3_5_chunk |                0 |               10
+ splitme         | _hyper_1_1_chunk | 1704326400000000 | 1704355200000000
+ splitme         | _hyper_1_3_chunk | 1704355200000000 | 1704643200000000
+ splitme         | _hyper_1_4_chunk | 1704643200000000 | 1704931200000000
+(4 rows)
+
+call split_chunk(:'int_chunk', split_at => '5');
```

Comment:
Add tests to split at `0`  as well as split at `10`? This would create an empty chunk, but should be good to test since it is an easy thing that can go wrong now or in the future.

Reply:
Splitting like that is not allowed. I can add tests for it.

Reply:
It would be great to have a test that this generates an error as well.